### ISSUE TITLE
Animania not working/No Forge BiomeDictionary Tags?

### ISSUE THREAD

**Issue by SirChamomile** [Conquest-Reforged/ConquestReforged#55]
The mod Animania replaces the vanilla mods with theirs, and yet when I play with both Reforged and Animania all of the mobs they overwrite are missing, as well as any that they add to the mod. From what I can tell they're spawning mobs using the BiomeDictionary: [https://github.com/capnkirok/animaniamod/blob/1.12/src/main/java/com/animania/common/handler/EntityHandler.java](Spawning)

Does Conquest Reforged use the forge BiomeDictionary? I was trying to look into other mods that might use this and see if they are working, but the one I could think of (Forestry), is a little esoteric for my skills. This one seems to be generating properly with the exception of the meadows hives. Wintry and Forest are generating fine.

Another mod I tried was Familiar Fauna, but this one wasn't working. Does Reforged affect animal spawns?

**Comment by HIPPOKLEIDES** [comment]
We do use BiomeDictionary and assign tags to our biomes. A few biome classes (mesa, ocean, river, red desert, beach, desert) had this.spawnableCreatureList.clear(); so perhaps that was the culprit. 
In all other cases we just add mobs to spawnableCreatureList.

I will test it myself when time permits.

**Comment by HIPPOKLEIDES** [comment]
note: doesn't seem like CR is working with Animania, however it does work with other mods adding mobs such as Ice and Fire

**Comment by SirChamomile** [comment]
I wasn't able to get familiar fauna to work, but I'm not sure if I checked to see if that one is hardcoded, but I'm pretty sure it isn't, as it's made by the biomes o plenty people.

**Comment by dags-** [comment]
@SirChamomile 
If it's CR coming in and undoing whatever Animania sets up during the game load process, then you might be able to work around it by renaming the Animania jar to, like, 'ZAnimania' - I think forge loads mods in alphabetical order on most OS's so this way you could persuade it load CR first

**Comment by MedievalPigeon213** [comment]
I think it's because there are so many plants with Conquest Reforged. Perhaps if Animania mobs could spawn on transparent blocks it might fix the issue. Try looking in desert biomes with CR and see if Animania mobs like jackrabbits and Nigeria goats spawn. If they do spawn then the transparent block issue might be correct.

**Comment by dags-** [comment]
Had nosey through Animania's source, think I follow what's going on:

1. Animania sets up it's animal spawns during the `preInit` phase by querying the BiomeDictionary for various biome tags.

2. ConquestReforged registers it's biomes to the BiomeDictionary during `preInit` (it's basically the earliest phase that we can do that).

If Animania loads before ConquestReforged (which it would due to alphabetically coming first) then it receives the preInit event before we do .: ConquestReforged biomes won't have been registered to the BiomeDictionary in time for Animania to use them for it's mob spawns.

#### Refs:
Animania 'CommonProxy':
https://github.com/capnkirok/animaniamod/blob/1.12/src/main/java/com/animania/proxy/CommonProxy.java

Animania 'EntityHandler':
https://github.com/capnkirok/animaniamod/blob/1.12/src/main/java/com/animania/common/handler/EntityHandler.java

____

### "But it works with BiomesOPlenty..."

Animania works with _some_ other biome-providing mods because it is specifically set to load after them if they're present.

See their dependency string in their main @Mod class
https://github.com/capnkirok/animaniamod/blob/1.12/src/main/java/com/animania/Animania.java#L43

____

### What's the fix?

I'm pretty sure Animania could register it's mob spawns in a later phase (`Init` or possibly even `PostInit`). This should make their spawns work with other mods that are not included in their dependency string too, so it's an all-round good idea really.

In summary, I'm fairly sure this is not CR's problem to fix but you might be able to work around it by renaming the Animania mod jar until they fix it (assuming you report the problem to them)

**Comment by SirChamomile** [comment]
I've tested, and confirmed that after changing the name of the mod, Animania works correctly. However, as @MedievalPigeon213 suggested, it does not spawn properly from all of the ground cover that CR adds. It would require Animania to both fire after Conquest Reforged and spawn over transparent blocks.

**Comment by SirChamomile** [comment]
Done, and posted over at Glitchfiend.
